x-pack/plugins/endpoint/public/applications/endpoint/store/hosts/host_pagination.test.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreStart, HttpSetup } from 'kibana/public';
+import { DepsStartMock, depsStartMock } from '../../mocks';
+import { AppAction, HostState, HostIndexUIQueryParams } from '../../types';
+import { Immutable, HostResultList } from '../../../../../common/types';
+import { History, createBrowserHistory } from 'history';
+import { hostMiddlewareFactory } from './middleware';
+import { applyMiddleware, Store, createStore } from 'redux';
+import { hostListReducer } from './reducer';
+import { coreMock } from 'src/core/public/mocks';
+import { urlFromQueryParams } from '../../view/hosts/url_from_query_params';
+import { uiQueryParams } from './selectors';
+import { mockHostResultList } from './mock_host_result_list';
+import { MiddlewareActionSpyHelper, createSpyMiddleware } from '../test_utils';
+
+describe('host list pagination: ', () => {
+  let fakeCoreStart: jest.Mocked<CoreStart>;
+  let depsStart: DepsStartMock;
+  let fakeHttpServices: jest.Mocked<HttpSetup>;
+  let history: History<never>;
+  let store: Store<Immutable<HostState>, Immutable<AppAction>>;
+  let queryParams: () => HostIndexUIQueryParams;
+  let waitForAction: MiddlewareActionSpyHelper['waitForAction'];
+  let actionSpyMiddleware;
+  const getEndpointListApiResponse = (): HostResultList => {
+    return mockHostResultList({ request_page_size: 1, request_page_index: 1, total: 10 });
+  };
+
+  let historyPush: (params: HostIndexUIQueryParams) => void;
+  beforeEach(() => {
+    fakeCoreStart = coreMock.createStart();
+    depsStart = depsStartMock();
+    fakeHttpServices = fakeCoreStart.http as jest.Mocked<HttpSetup>;
+    history = createBrowserHistory();
+    const middleware = hostMiddlewareFactory(fakeCoreStart, depsStart);
+    ({ actionSpyMiddleware, waitForAction } = createSpyMiddleware<HostState>());
+    store = createStore(hostListReducer, applyMiddleware(middleware, actionSpyMiddleware));
+
+    history.listen(location => {
+      store.dispatch({ type: 'userChangedUrl', payload: location });
+    });
+
+    queryParams = () => uiQueryParams(store.getState());
+
+    historyPush = (nextQueryParams: HostIndexUIQueryParams): void => {
+      return history.push(urlFromQueryParams(nextQueryParams));
+    };
+  });
+
+  describe('when a new page size is passed', () => {
+    beforeEach(() => {
+      historyPush({ ...queryParams(), page_size: '20' });
+    });
+    it('should modify the url correctly', () => {
+      expect(queryParams()).toMatchInlineSnapshot(`
+        Object {
+          "page_index": "0",
+          "page_size": "20",
+        }
```

Comment:
@parkiino - are these values the defaults we are using? It would be good to test the defaults for this test case and have a separate test case for invalid values that revert to defaults.

Reply:
the defaults are going to be page_index: 0, and page_size: 10